src/components/protocol_handler/src/protocol_packet.cc

```diff
@@ -324,29 +345,17 @@ RESULT_CODE ProtocolPacket::ProtocolHeaderValidator::validate(
   switch (header.frameType) {
     case FRAME_TYPE_SINGLE:
     case FRAME_TYPE_CONSECUTIVE:
+    case FRAME_TYPE_CONTROL:
```

Comment:
@KVGrygoriev Are you sure that when `FRAME_TYPE_CONTROL` the  `header.dataSize` shouldn't be a zero? When it was checked in separate case it could be.

Reply:
@ZhdanovP good point, see e9e7728e3c0b11328d55b083a8bfcbd1542c1f89